pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -173,18 +174,24 @@ func (c *Controller) Services() []*model.Service {
 			services = append(services, svcs...)
 		} else {
 			for _, s := range svcs {
-				sp, ok := smap[s.Hostname]
+				previous, ok := smap[s.Hostname]
 				if !ok {
 					// First time we see a service. The result will have a single service per hostname
 					// The first cluster will be listed first, so the services in the primary cluster
 					// will be used for default settings. If a service appears in multiple clusters,
 					// the order is less clear.
-					smap[s.Hostname] = s
+					smap[s.Hostname] = index
```

Comment:
What is the use of this index. It is not clear to me?

Reply:
The function is appending a services slice,  the index is used to record the index of a svc, so when I need to merge this svc with a new svc, I can update the slice quicky

Reply:
i see